src/context/jwtFunctions.ts

```diff
@@ -0,0 +1,27 @@
+import {jwtDecode} from 'jwt-decode';
+import {JWTData} from "../types/types.ts";
+
+export const decodeToken = (token: string | null): JWTData | null => {
+    if (!token) return null;
```

Comment:
I don't like returning null in most cases, I would instead throw error for !token and move try catch to parent scope. You don't necessarily need to catch errors everywhere, rule of thumb is to always try catch at highest level possible of execution